src/core/execute.c

```diff
@@ -3124,6 +3126,48 @@ static int compile_bind_mounts(
         return r;
 }
 
+/* ret_symlinks will contain a list of pairs src:dest that describes
+ * the symlinks to create later on. */
```

Comment:
Can you extend these, i.e. say that these are the symlinks /var/lib/xyz → /var/lib/private/xyz which are needed to safely give private dirs to DynamicUser=1 users.